### PR TITLE
Introduction of mocktails and milkshake expansion.

### PR DESCRIPTION
**Pull request by carshalash**
## About The Pull Request

Quite a few people ignore going to the bar because they like to have their character not drink (Let's face it, 80% of the player base is getting lightdrinker for quirk points, not RP).  At the moment the 4 accessible options for non-alcoholic drinks are Gibb float, arnold palmer, lemonade and triple citrus; I honestly think that triple citrus shouldn't count though but whatever.

image

From left to right

**Cinderella**: Made to look identical to a mocktail, a nice blend of fruits topped off with ginger ale and a few dashes of angostura to trick party-goers into thinking you're still drinking.
**Agua fresca**: Went with a watermelon agua fresca, which refreshes your insides and lowers your body temp. 
**Roy Rogers**: Cola with grenadine in it, maybe nice at first but all that sugar will make you shake
**Shirley temple**: Ginger ale mixed with grenadine, quite nice but not as nice on the stomach as a normal Sol Dry™ 
**Vanilla shake**: Basic milkshake, personally my favourite when mint isn't available. Had to restrain myself from making mint shakes due to bloat.
**Choccy shake**: For some reason everyone's favourite flavour. It was hard not to be a contrarian when writing the flavor text for this one.
**Caramel shake**: Has an increased amount of nutriment compared to the other shakes, the downside is a reduced drink quality due to the sweetness.

I hope you all like these drinks and make them in the bar going forward! 

## Why It's Good For The Game

I said so.

More options are never a bad thing, I tried to give the drinks some kind of purpose so they aren't just bloat+flavor and kept things within balance. Again, as shown in the introduction I believe a lot of people don't see much purpose in the bar if they can't drink due to quirks/rp purpose, hopefully, this will make it seem like a valid option to spend time.

I think the opportunity for there to be racist old fashioned milkshake shops filled with greasers that bully lizards over their inability to process dairy is a good enough reason for this PR to be merged. 

## Changelog

:cl:
expansion: Introduces new non-alcoholic beverages to be made at the bar!
imageadd: Sprites for these new beverages! Not just coloured glass
/:cl:

### CODE REVIEW

**Comment by EOBGames** [comment]
Cola and grenadine- please tell me that's not a real drink, that sounds like drinking pure sugar

**Comment by williamcll** [comment]
Would be cool if there's boiling hot ginger cola.

**Comment by carshalash** [comment]
A CHALLENGER APPROACHES

image

**Italian Hot Chocolate**: Tired of that watery stuff from the machines? Have some molto bene Italian hot chocolate while relaxing and reading a nice book in the library. Certified to make you feel happier and fuller than drinking the regular stuff.

**Comment by thestubborn** [comment]
Add the Doctails. https://venturebrothers.fandom.com/wiki/Doctails
They heal scientists but make everyone else puke.

**Comment by Sealed101** [comment]
wasnt there shake variants already or am i delusional

**Comment by ATH1909** [comment]
> A CHALLENGER APPROACHES
> 
> image
> 
> **Italian Hot Chocolate**: Tired of that watery stuff from the machines? Have some molto bene Italian hot chocolate while relaxing and reading a nice book in the library. Certified to make you feel happier and fuller than drinking the regular stuff.

add this to the OP

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -900,6 +971,26 @@
...
+	taste_description = "thick creamy chocolate"
+	glass_icon_state = "italiancoco"
+	glass_name = "glass of italian coco"
+	glass_desc = "A spin on a winter favourite, made to please."
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/consumable/italian_coco/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	M.adjust_bodytemperature(5 * REM * TEMPERATURE_DAMAGE_COEFFICIENT * delta_time, 0, M.get_body_temp_normal())
+	if(M.getBruteLoss() && DT_PROB(10, delta_time))
+		M.heal_bodypart_damage(1, 0, 0)
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:988]: You know that not every single drink needs to have a unique effect right? specially when it's just a random healing one.

If you can't think of anything more intriguing or cant' be bothered coding up something unique then just don't add an effect, don't just fall back on some lazy ass tiny healing

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -1066,6 +1172,24 @@
...
+	taste_description = "cool refreshing watermelon"
+	glass_icon_state = "aguafresca"
+	glass_name = "Agua Fresca"
+	glass_desc = "90% water, but still refreshing."
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/consumable/agua_fresca/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	M.adjust_bodytemperature(-8 * REM * TEMPERATURE_DAMAGE_COEFFICIENT * delta_time, M.get_body_temp_normal())
+	if(M.getToxLoss() && DT_PROB(10, delta_time))
+		M.adjustToxLoss(-1, 0)
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1189]: you did it again

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (carshalash) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:988]: Inherits stats from hot coco, it's meant to be a flat upgrade to it.

Comment (carshalash) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:988]: If you want the brute part to be gone then I can budge on this one.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (carshalash) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1189]: I would really like this one to retain toxloss, I want it to be really refreshing and light in every sense.

**Comment by optimumtact** [comment]
i just think it's silly to have so many effects that do the sam ething

**Comment by anconfuzedrock** [comment]
> A CHALLENGER APPROACHES
> 
> image
> 
> **Italian Hot Chocolate**: Tired of that watery stuff from the machines? Have some molto bene Italian hot chocolate while relaxing and reading a nice book in the library. Certified to make you feel happier and fuller than drinking the regular stuff.

is that hot coco with spaghetti and fucking meatballs in it

**Comment by optimumtact** [comment]
no dude it's proper coffee

**Review by Twaticus — APPROVED** [review]

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
on mob life needs return values, you can prob get away with return ..() for most of these.

Also do not make more drinks after this, nonalcoholic or otherwise (unless you plan to replace, in which case they should have unique functionality).

**Comment by carshalash** [comment]
> on mob life needs return values, you can prob get away with return ..() for most of these.
> 
> Also do not make more drinks after this, nonalcoholic or otherwise (unless you plan to replace, in which case they should have unique functionality).

I think you should take a lil peak at the drinks reagent dm. Most life code doesn't have a return on it.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (carshalash) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Drinks check for toxloss.

**Pull request merged**